### ISSUE TITLE
Content and Background Scripts are not executing in dev mode

### ISSUE THREAD

**Issue by theharshin** [adambullmer/vue-cli-plugin-browser-extension#63]
I've installed the plugin (v0.22.5) on the fresh setup of Vue CLI and the serve command builds the files successfully. The extension loads and displays 
`[ WER: Connected to Extension Hot Reloader ]` but code in content-script or background is not being executed. 
Surprisingly, when I build the extension and load it, everything works fine! Something to do it Extension Hot Reloader?

**Comment by theharshin** [comment]
I've tried to use v0.19.0 with the same setup and everything works fine! 🤔

**Comment by theharshin** [comment]
<img width="838" alt="image" src="https://user-images.githubusercontent.com/7118937/67160382-c2ff3200-f36d-11e9-9fa1-56e220454c4d.png">

After removing this check, everything works fine.

**Comment by rrfaria** [comment]
I opened a PR suggesting this change 
https://github.com/adambullmer/vue-cli-plugin-browser-extension/pull/64

**Comment by adambullmer** [comment]
Can you share you `vue.config.js` file and your `src/manifest.json`? Also how are you trying to invoke your content scripts and background scripts?

**Comment by rrfaria** [comment]
I found out the problem is not in the code please discart this pr.

the plugin even you are selecting content-script in options,  it doesn't add on `src/manifest.json` and only `vue.config.js` is right

```
"content_scripts": [
    {
      "matches": ["*://*/*"],
      "js": ["js/content-script.js"]
    }
  ],

```

**Comment by adambullmer** [comment]
Correct. The snippet you added automatically injects a content script into every page. It is my opinion that this behavior should not be the default scaffolding behavior of browser extensions, as most developers will probably never circle back and change this to a narrower list if they didn't need it in the first place.

That being said, if this continues to be a pain point, then this means there is a difference in expectations of this plugin between myself and the community, or a lapse in documentation around this particular feature. Probably a good first step is to highlight this fact in the readme and see if there continues to be issues around expecting automatically injected content scripts on every page.

**Comment by rrfaria** [comment]
@adambullmer I believe the misunderstood is happen because when you are creating your first extension there are some options you can choose and when you select "content-script" between them we were expecting no need to setup more things to run it. 

sorry for that.

**Comment by JasonLearmouth** [comment]
Hello @adambullmer, I'm having a similar issue where my background.js code is not running. I have the default content (generated by your extension) in there, so I was expecting the content script to log a message in the web page console, and the background script to log a message to the background console. Neither of these is happening. 

Looking at the conversation above, it seems like I need to add some config to get it working, but I don't have enough experience to know what to add or where to add it.

Can you clarify the steps required to get background and content scripts working with the out-of-the-box template?

**Comment by adambullmer** [comment]
I think I finally stumbled on the root cause of content scripts and background scripts not executing, and should be released in `0.23.1`. It looks like vue added in some optimizations in development mode, but only if your project were using a more recent version of vue-cli-service, and would explain why this was a little harder to track down.

@JasonLearmouth concerning your troubles, I think there is a bug in the scaffolding where the incorrect listener is being applied on the background file when also scaffolding with a popup (the default 2 options). This should be fixed in `0.23.2`. Depending on how much work you've put into your project I may recommend just rescaffolding with `vue invoke browser-extension` once you've updated to the latest version. Otherwise, you can replace the top line of your backgound.js file with this: 
```js
browser.runtime.onMessage.addListener(function (request, sender, sendResponse) {
```

**Comment by tnishimura** [comment]
@adambullmer FYI I don't think this issue is entirely fixed. I just fresh 'vue create' and 'vue add browser-extension', and in dev mode, the background script and content scripts are not running. My package versions are: 

```
yarn list v1.12.3
├─ @vue/babel-helper-vue-jsx-merge-props@1.0.0
├─ @vue/babel-plugin-transform-vue-jsx@1.1.2
├─ @vue/babel-preset-app@4.3.0
├─ @vue/babel-preset-jsx@1.1.2
├─ @vue/babel-sugar-functional-vue@1.1.2
├─ @vue/babel-sugar-inject-h@1.1.2
├─ @vue/babel-sugar-v-model@1.1.2
├─ @vue/babel-sugar-v-on@1.1.2
├─ @vue/cli-overlay@4.3.0
├─ @vue/cli-plugin-babel@4.3.0
├─ @vue/cli-plugin-eslint@4.3.0
├─ @vue/cli-plugin-router@4.3.0
├─ @vue/cli-plugin-vuex@4.3.0
├─ @vue/cli-service@4.3.0
├─ @vue/cli-shared-utils@4.3.0
├─ @vue/component-compiler-utils@3.1.1
├─ @vue/preload-webpack-plugin@1.1.1
├─ @vue/web-component-wrapper@1.2.0
├─ eslint-plugin-vue@6.2.2
├─ vue-cli-plugin-browser-extension@0.24.0
├─ vue-eslint-parser@7.0.0
├─ vue-hot-reload-api@2.3.4
├─ vue-loader@15.9.1
├─ vue-style-loader@4.1.2
├─ vue-template-compiler@2.6.11
├─ vue-template-es2015-compiler@1.9.1
└─ vue@2.6.11
```
